### PR TITLE
Initial telemetry extension

### PR DESCRIPTION
**Pull request by jakolehm**
- refactor api from "store.register" to "registry.add" pattern
- fix issues with stores
- autoload only extensions that are configured in Lens package.json

### CODE REVIEW

File: src/renderer/bootstrap.tsx [hunk 1]
```diff
@@ -19,7 +20,8 @@ type AppComponent = React.ComponentType & {
 export {
   React,
   Mobx,
-  LensExtensions
+  LensExtensions,
```
Comment (ixrock) [inline src/renderer/bootstrap.tsx:23]: Why do we need to export it here? Where it's used afterwards?

**Review by ixrock — APPROVED** [review]
LGTM

**Review by aleksfront — APPROVED** [review]

File: src/renderer/bootstrap.tsx (hunk 1 above)
Comment (jakolehm) [reply src/renderer/bootstrap.tsx:23]: `LensExtensions` includes store related APIs which are needed also in renderer.

Comment (ixrock) [reply src/renderer/bootstrap.tsx:23]: Actually I mean whole `export` from `bootstrap.tsx`. This is entry point for renderer processes, just curious how it's consumed?

Comment (jakolehm) [reply src/renderer/bootstrap.tsx:23]: All extension apis are exported from `src/main/index.ts` (for main extension api) and `src/renderer/bootstrap.tsx` (for renderer extension api) using webpack (webpack allows to export modules from an entrypoint). Extensions can consume these via webpack externals.

Comment (ixrock) [reply src/renderer/bootstrap.tsx:23]: Hmm. I thought using/marking webpack's externals as global variable should be enough..
Is it documented somewhere in webpack's docs that entry point should export actually something in addition?

Comment (jakolehm) [reply src/renderer/bootstrap.tsx:23]: https://webpack.js.org/configuration/output/#outputlibrarytarget

Comment (ixrock) [reply src/renderer/bootstrap.tsx:23]: Ok, thanks!

**Pull request merged**